repose-aggregator/src/docs/asciidoc/filters/keystone-v2.adoc

```diff
@@ -1,52 +1,413 @@
+* `X-Impersonator-Roles` - Identifies roles of the impersonator. (ex. "racker", "admin")
+
+The Keystone v2 Identity service also has other attributes it provides when a token is validated.
+If any of this information is provided, then it will be passed in the following headers:
+
+* `X-Contact-ID` - The Contact ID for the user.
+* `X-Default-Region` - The Default Region for the user.
+* `X-Tenant-ID` - The Tenant ID's for the user.
+** The value of this header is governed by `send-all-tenant-ids` attribute and what is provided by the Keystone v2 Identity service.
+* `X-Tenant-Name` - The Tenant Name for the user.
+* `X-Token-Expires` - The date/time of when the token provided by the Keystone v2 Identity service expires.
+
+If delegation is enabled, then the `X-Delegated` header is created.
+This is mainly intended for use by the <<herp.adoc#, Highly Efficient Record Processor (HERP) filter>> and <<derp.adoc#, Delegation Response Processor (DeRP) filter>> for internal delegation processing within Repose.
+However, it can be exposed to the origin service under certain configurations.
 
 === Request body changes
-???.
+This filter does not modify the request body.
 
 === Recommended follow-on (succeeding) filters
-???.
+While there are no succeeding filters that are strictly required, the following filters may be useful:
+
+* <<simple-rbac.adoc#, Simple RBAC filter>> - Provides role-based access control to the origin service's API, making use of the `X-PP-Groups` header.
+* <<api-validator.adoc#, API Validator filter>> - Provides role-based access control to the origin service's API, making use of the `X-PP-Groups` header.
+* <<rate-limiting.adoc#, Rate Limiting filter>> - Provides rate limiting, making use of the `X-PP-User` header.
 
 === Response body changes
-???.
+This filter does not modify the response body.
 
 === Response headers created
-???.
+* `Retry-After` - This is included on all _Service Unavailable_ (`503`) responses to indicate when it is appropriate to retry the request again.
+* `WWW-Authenticate` - This is included on all _Unauthorized_ (`401`) responses to challenge the authorization of a user agent.
 
 === Response Status Codes
-???.
+[cols="a,a,a,a", options="header"]
+|===
+|When the Keystone v2 Identity service returns:
+|Repose Get Admin Token Call Returns
+|Repose Validate Token Call Returns
+|Repose Groups Call Returns
+
+| _Successful_ (`2xx`)
+| _OK_ (`200`)
+| _OK_ (`200`)
+| _OK_ (`200`)
+
+| _Bad Request_ (`400`)
+| _Internal Server Error_ (`500`)
+| _Internal Server Error_ (`500`)
+| _Internal Server Error_ (`500`)
+
+| _Unauthorized_ (`401`)
+
+* The admin token has expired.
+| _Internal Server Error_ (`500`)
+| _Internal Server Error_ (`500`)
+| _Internal Server Error_ (`500`)
+
+| _Unauthorized_ (`401`)
+
+* Self-validating tokens are being used, and the user token has expired.
+|
+|_Unauthorized_ (`401`)
+|_Unauthorized_ (`401`)
+
+| _Payment Required_ (`402`)
```

Comment:
Confirmed. There is no circumstance that Keystone v2/Identity returns a `402`. Removed.